src/routes/contents/[id]/+page.svelte

```diff
@@ -0,0 +1,205 @@
+                break;
+            case 'screen':
+                //goes to another contents page
+                contentsStack.pushItem($page.data.menu.id);
+                await goto(`${base}/contents/${item.linkTarget}`);
+                break;
+            case 'other':
+                //switch on item.linkLocation
+                switch (item.linkLocation) {
+                    case 'about':
+                    case 'settings':
+                        goto(`${base}/${item.linkLocation}`);
+                        break;
+                    case 'layout':
+                        modal.open(MODAL_COLLECTION);
+                        break;
+                    case 'website':
+                        //opens in a separate tab
+                        window.open(`${item.linkTarget}`, '_blank');
+                        break;
+                }
+                break;
+
+            default:
+                // For other book types (e.g. quiz), the linkType will be
+                // the book type and the linkLocation will have the route
+                // to the viewer of the book type.
+                goto(`${base}/${item.linkLocation}`);
+                break;
+        }
+    }
+
+    function setReference(item) {
+        let book;
+        let chapter;
+        let verse;
+        const reference = item.linkTarget.split('.');
+
+        switch (reference.length) {
+            case 1:
+                book = reference[0];
+                break;
+            case 2:
+                book = reference[0];
+                chapter = reference[1];
+                break;
+            case 3:
+                book = reference[0];
+                chapter = reference[1];
+                verse = reference[2];
+                break;
+            default:
+                break;
+        }
+        refs.set({
+            book: book,
+            chapter: chapter,
+            verse: verse
+        });
+    }
+
+    //set the title for the current contents page
+    function setTitle(page) {
+        //checks title type and returns the appropriate title or lack of title
+        let title = '';
+        switch (page.data.features['title-type']) {
+            case 'app-name':
+                title = config.name;
+                break;
+            case 'screen':
+                title = page.data.menu.title[$language] ?? page.data.menu.title.default ?? '';
+                break;
+            case 'custom':
+                title = page.data.title[$language] ?? page.data.title.default ?? '';
+                break;
+            case 'none':
+            default:
+                break;
+        }
+        return title;
+    }
+
+    function handleBackNavigation(event) {
+        event.preventDefault();
+        if ($contentsStack.length > 0) {
+            const menuId = contentsStack.popItem();
+            goto(`${base}/contents/${menuId}`);
+        }
+    }
+    $: showBackButton = $contentsStack.length > 0;
```

Comment:
This one is correct.  In the contents menu itself, the back arrow only shows if the stack is greater than zero.